File: my-folder/0011-container-with-most-water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left =0;
        int right = height.size()-1;
        int output = 0;

        while (left<right)
        {
            int min_height = min(height[left], height[right]);
            int diff = abs(right - left);
            output = max(output, (min_height)*diff);

            if (height[left]<=height[right])
            {
                left++;
            }
            else
            {
                right--;
            }
            
        }
        return output;
        
    }
};

File: my-folder/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:

        n = len(height)
        left = 0
        right = n - 1

        ans = float('-inf')
        while left < right:
            current_area = min(height[left], height[right]) * (right - left)
            ans = max(ans, current_area)

            if height[left] > height[right]:
                right-=1
            else:
                left+=1
        
        return ans

        

        

